src/object/srv_obj_migrate.c

```diff
@@ -2567,8 +2567,13 @@ ds_migrate_fini_one(uuid_t pool_uuid, uint32_t ver)
 	ABT_mutex_lock(tls->mpt_inflight_mutex);
 	ABT_cond_broadcast(tls->mpt_inflight_cond);
 	ABT_mutex_unlock(tls->mpt_inflight_mutex);
+
 	migrate_pool_tls_put(tls); /* lookup */
+	ABT_eventual_wait(tls->mpt_done_eventual, NULL);
+
 	migrate_pool_tls_put(tls); /* destroy */
+
+	D_DEBUG(DB_TRACE, "migrate fini one ult "DF_UUID"\n", pool_uuid);
```

Comment:
Another issue I realized is that in migrate_cont_iter_cb(), it compares "tls->mpt_obj_generated_ult" with "tls->mpt_obj_executed_ult" for ULT throttling, but the tls is on xstream 0, so the 'tls->mpt_obj_executed_ult' should always be zero...

Same problem to the 'mpt_generated_ult' and 'mpt_executed_ult' comparing.

Reply:
ds_migrate_query_status will update the xstream 0 status.

Reply:
Then how the throttling on 'mpt_generated_ult' work? It's increased on VOS xstream tls, and xstream 0 doesn't know how many ults are already generated until next ds_migrate_query_status() call, right?

Reply:
ds_migrate_query_status will be called every 2 seconds automatically for reporting the migration status to the leader, at that time xstream 0 status will be updated.

Reply:
Right, I mean there could be much more ULTs than the limited number being created during the 2 seconds. Anyway, I think it's not related to ticket, we could cleanup it later.